extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_generation.py

```diff
@@ -105,19 +117,20 @@ def construct_stream_output(
         options (InferenceOptions): The inference options. Used to determine the stream callback.
 
     """
-    accumulated_message = ""
     output = ExecuteResult(
             **{
                 "output_type": "execute_result",
-                "data": accumulated_message,
+                "data": "", # We update this below
                 "execution_count": 0, #Multiple outputs are not supported for streaming
                 "metadata": {},
             }
         )
+    accumulated_message = ""
     for new_text in streamer:
-        accumulated_message += new_text
-        options.stream_callback(new_text, accumulated_message, 0)
-        output.data = accumulated_message
+        if isinstance(new_text, str):
```

Comment:
Can this be anything other than string? These changes seem unnecessary if I'm understanding the streamer correctly

Reply:
Streamer can actually contain anything, doesn't have to be text. Theoretically it always be str (except for the stop signal), but I'm just being safe. This is a fork of the HF source code (couldn't find it publicly on GH, but it's this: https://github.com/PaddlePaddle/PaddleNLP/blob/a55039cc16bbcfa06b204a5f21b813e98a3f65ca/paddlenlp/generation/streamers.py#L197-L206)